src/UnifiedFFI/ExternalData.extension.st

```diff
@@ -46,7 +47,10 @@ ExternalData >> readStringUTF8 [
 	[(char := handle unsignedByteAt: index) = 0 ] whileFalse: [
 		stream nextPut: char.
 		index := index + 1].
-	^ ZnCharacterEncoder utf8 decodeBytes: stream contents
+	^ [ ZnCharacterEncoder utf8 decodeBytes: (bytes := stream contents) ]
```

Comment:
just. wondering if it is not better to get the exception to know that the conversion failed instead of getting a string with wrong content. In which case is it better to ignore the decoding error?

Reply:
Hi @demarey,

Good point, although I would provide two methods in ExternalData, one which is strict UTF8 and the other forgiving.  The caller can then decide which to use (instead of every caller having to write their own exception handler).

I'll update the PR (tomorrow).

Thanks!
Alistair

Reply:
Hi Alistair,
Yes, looks a good plan. Thanks!

Reply:
Hi @demarey,

Thanks again for your feedback, but after looking at this further, I don't think creating a forgiving version of #readStringUTF8 is the correct approach.  As you imply, the application should be making the decision about whether the UTF8 encoding is important, and there are too many layers between the application (Iceberg in the case) and FFI for this to work.

I've opened a separate issue to suggest a new approach - https://github.com/pharo-project/pharo/issues/4687

Cheers,
Alistair